addons/survey/models/survey_survey.py

```diff
@@ -378,6 +380,12 @@ def _onchange_survey_type(self):
                 'scoring_type': 'scoring_with_answers',
             })
 
+    @api.constrains('scoring_type', 'users_can_go_back')
+    def _check_scoring_after_page_availability(self):
+        for survey in self:
+            if survey.scoring_type == 'scoring_with_answers_after_page' and survey.users_can_go_back:
+                raise ValidationError(_("You cannot use roaming when scoring type is scoring with answers after each page."))
```

Comment:
In cases like this, we try to run the check on all surveys of the recordset and return a message referencing all records: like having 
> Please review the configuration of the following surveys:
> - survey 1
> - Survey 2
You should find examples in our apps: event, eLearning,...